src/core/annotation.js

```diff
@@ -1824,89 +1829,115 @@ class WidgetAnnotation extends Annotation {
       rotation = this.rotation;
     }
 
-    let appearance = await this._getAppearance(
-      evaluator,
-      task,
-      annotationStorage
-    );
-    if (appearance === null) {
-      return null;
+    let appearance = null;
+    if (!this._needAppearances) {
+      // No need to create an appearance: the pdf has the flag /NeedAppearances
+      // which means that it's up to the reader to produce an appearance.
```

Comment:
Doesn't the condition and comment contradict each other, since `this._needAppearances === false` is the default value when the /NeedAppearances flag *does not* exist?